txscript/script.go

```diff
@@ -446,23 +446,22 @@ func calcWitnessSignatureHash(subScript []parsedOpcode, sigHashes *TxSigHashes,
 	binary.LittleEndian.PutUint32(bIndex[:], txIn.PreviousOutPoint.Index)
 	sigHash.Write(bIndex[:])
 
-	if isWitnessPubKeyHash(subScript) {
+	if isWitnessPubKeyHashScript(scriptSig) {
```

Comment:
`subScript` here is the witness program script, this trick is used to check if it's a `p2wkh` script, then "inject" the proper witness script into the sighash computation. So this logic is still correct, but the naming is off.